plugins/harness-ccm/src/hooks/useResourceSlugFromEntity.tsx

```diff
@@ -0,0 +1,46 @@
+import { match } from 'path-to-regexp';
+
+export const useResourceSlugFromEntity = (perspectiveUrl?: string) => {
+  const cleanedString = perspectiveUrl?.split(' ')[0];
+  let accountId;
+  let orgId;
+  let projectId;
+  let perspectiveId;
+
+  const urlMatch = match(
```

Comment:
@rajarshee-harness here we need to add all the URLs that are available for perspective ID, like new nav old nav unified view, example https://github.com/harness/backstage-plugins/blob/41c06019d6ce9c3991322956c766342103db8354/plugins/harness-ci-cd/src/hooks/useProjectSlugEntity.tsx#L25-L67